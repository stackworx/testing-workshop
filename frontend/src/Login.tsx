import React, {Component} from 'react';
import {Formik, Form, Field, ErrorMessage} from 'formik';
import * as Yup from 'yup';

const validationSchema = Yup.object().shape({
  email: Yup.string()
    .email()
    .required(),
  password: Yup.string()
    .min(6, 'Password has to be longer than 6 characters!')
    .required(),
});

interface Props {
  handleSubmit: (opts: {email: string; password: string}) => Promise<void>;
}

export class Login extends Component<Props> {
  render() {
    const {handleSubmit} = this.props;

    return (
      <div style={{padding: 20}}>
        <Formik
          validationSchema={validationSchema}
          initialValues={{email: '', password: ''}}
          onSubmit={async (values, {setSubmitting}) => {
            try {
              handleSubmit(values);
              setSubmitting(false);
            } catch (ex) {
              setSubmitting(false);
            }
          }}
        >
          {({isSubmitting}) => (
            <Form>
              Email:
              <Field type="email" name="email" /> <br />
              <ErrorMessage name="email" component="div" />
              <br />
              Password:
              <Field type="password" name="password" />
              <br />
              <ErrorMessage name="password" component="div" />
              <br />
              {isSubmitting && <div>Submitting...</div>}
              <button type="submit" disabled={isSubmitting}>
                Submit
              </button>
            </Form>
          )}
        </Formik>
      </div>
    );
  }
}
