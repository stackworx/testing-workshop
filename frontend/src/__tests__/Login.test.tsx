import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from 'react-testing-library';

import 'jest-dom/extend-expect';
import {Login} from '../Login';

// automatically unmount and cleanup DOM after the test is finished.
// @ts-ignore
afterEach(cleanup);

test('renders correctly', async () => {
  const tree = renderer
    .create(<Login handleSubmit={async () => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('test required fields', async () => {});

test('test invalid email', async () => {});

test('test invalid password', async () => {});

test('test valid submit', async () => {});
