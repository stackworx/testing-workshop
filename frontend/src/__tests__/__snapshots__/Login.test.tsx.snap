// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "padding": 20,
    }
  }
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    Email:
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="email"
      value=""
    />
     
    <br />
    <br />
    Password:
    <input
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <br />
    <button
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
